common/persistence/history_manager.go

```diff
@@ -127,13 +123,18 @@ func (m *executionManagerImpl) ForkHistoryBranch(
 		return nil, err
 	}
 
-	branchInfoBlob, err := m.serializer.HistoryBranchToBlob(newBranchInfo, enumspb.ENCODING_TYPE_PROTO3)
```

Comment:
I probably lack some context on what needs to be customized here. But `m.serializer` is something that can be changed during initialization (search for `FactoryProvider` in persistence/factory/fx.go of your repo). Can that be used to achieve your goal?

Reply:
Synced offline. Currently here's no way to inject serializer for the one used in ndcHistoryReplicator, so have to do it this way for now. 

Once ^ is possible, then I think we should go with that route.

Reply:
cc @yiminc What do you think?